common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -281,6 +304,8 @@ def transcript(self, request, dispatch):
                 transcript_content, transcript_filename, transcript_mime_type = self.get_transcript(
                     transcripts, transcript_format=self.transcript_download_format, lang=lang
                 )
+            except (KeyError, UnicodeDecodeError):
+                return Response(status=404)
```

Comment:
This looks unrelated?

Reply:
I moved it up from below while debugging something because it turns out UnicodeDecodeError is a subclass of ValueError, so that case was getting swallowed by the other `except` handler block below.